core/data_encoder_decoder.py

```diff
@@ -134,9 +131,6 @@ def decode(self, encode_reader: EncodedBlockReader, output_stream: DataStream):
             output_stream (DataStream): DataStream object to write decoded blocks of data
         """
 
-        # reset the state
```

Comment:
The reason to call `reset` at the beginning of `encode` and `decode` functions was so that when we call the same 
`Encoder.encode` on another input, the state maintained is re-freshed. 

As you remove the call to reset, the onus is on the user to reset the encoder (in case the user wants to). This is probably better. Can you fix the comments in the `reset` function.. as it is still saying in the NOTE that `reset` functions gets called before encode/decode.. maybe say that the user can call reset if he/she wants to